### PR TITLE
Implemented Enumerable.any and .all with tests.

### PR DESCRIPTION
**Pull request by wseidel**
Hi!
I have implemented some methods and I want implement some more. 
I'm sending these to be evaluated. 

Hugs.

### CODE REVIEW

File: src/test/java/net/hydromatic/linq4j/test/Linq4jTest.java [hunk 1]
```diff
@@ -182,6 +183,48 @@ public boolean apply(Department v1) {
     assertEquals(2, count);
   }
 
+  @Test public void testAllPredicate() {
+
```
Comment (julianhyde) [inline src/test/java/net/hydromatic/linq4j/test/Linq4jTest.java]: we don't generally start a method with a blank line

File: src/main/java/net/hydromatic/linq4j/EnumerableDefaults.java [hunk 2]
```diff
@@ -114,7 +122,22 @@ public static boolean any(Enumerable enumerable) {
    */
   public static <TSource> boolean any(Enumerable<TSource> enumerable,
       Predicate1<TSource> predicate) {
-    throw Extensions.todo();
+
+    boolean result = false;
```
Comment (julianhyde) [inline src/main/java/net/hydromatic/linq4j/EnumerableDefaults.java]: i wouldn't use a variable - just return true or false as soon as you know the answer

**Comment by julianhyde** [comment]
I've made a couple of comments about cosmetic issues. Fix these and I'll merge. 

Thanks for the pull request, looking forward to more.

PS I don't think the travis failure on jdk 1.7 is your fault.

**Comment by wseidel** [comment]
Cosmetics: done.

**Comment by julianhyde** [comment]
Rebased as https://github.com/julianhyde/linq4j/commit/2ea04d5d7daae8a881391c81cc9b6e83fceceaf4.

Protip - I always create a new branch for each pull request rather than using "master". Less confusing for you and for me.

**Comment by wseidel** [comment]
Thanks for the tip. I'll be more careful. ;)

**Pull request closed without merging**